robowflex_library/include/robowflex_library/adapter.h

```diff
@@ -3,18 +3,19 @@
 #ifndef ROBOWFLEX_ADAPTER_
 #define ROBOWFLEX_ADAPTER_
 
-#include <Eigen/Core>
-#include <Eigen/Geometry>
+#include <type_traits>
 
-#include <robowflex_library/macros.h>
+#include <moveit/transforms/transforms.h>
 
 namespace robowflex
 {
-#if ROBOWFLEX_MOVEIT_ISOMETRY
-    using RobotPose = Eigen::Isometry3d;
-#else
-    using RobotPose = Eigen::Affine3d;
-#endif
+    using RobotPose = std::remove_cv<                                      //
+        std::remove_reference<                                             //
+            decltype(                                                      //
+                std::declval<moveit::core::Transforms>().getTransform("")  //
+                )                                                          //
+            >::type                                                        //
+        >::type;
```

Comment:
Nice, maybe use std::decay instead? Also, why the empty comments at the end of the line?

Reply:
Good suggestion with `std::decay`, will push in a second

Reply:
Empty comments are to make `clang-format` not blow up the expanded type declaration for readability

Reply:
Just FYI, you could also do this:
```
// clang-format off
...
// clang-format on
```
What you have is fine. I just didn't understand at first.

Reply:
The nice bit with just adding empty line comments is that `clang-format` will still do in-line alignment and formatting, and not just give up the ghost entirely.